src/util/hass-translation.html

```diff
@@ -2,24 +2,13 @@
 
 <script>
 function getActiveTranslation() {
-  // Perform case-insenstive comparison since browser isn't required to
-  // report languages with specific cases.
-  const lookup = {};
-  /* eslint-disable no-undef */
-  Object.keys(window.translationMetadata).forEach((tr) => {
-    lookup[tr.toLowerCase()] = tr;
-  });
-
-  // Search for a matching translation from most specific to general
   function languageGetTranslation(language) {
-    const subtags = language.toLowerCase().split('-');
-
-    for (let i = subtags.length; i >= 1; i--) {
-      const lang = subtags.slice(0, i).join('-');
-
-      if (lookup[lang]) {
-        return lookup[lang];
-      }
+    if (window.translationMetadata[language]) {
+      return language;
+    }
+    if (language.substring(0, 2) === 'zh') {
```

Comment:
We should spilt on the dash here as well, so we aren't matching three letter language codes that begin with `zh`.